### PR TITLE
Refactoring pruner to use AP Scheduler

### LINKED ISSUE #891: Add file to pruner test
A prune task was added to pruner to delete files. The new task should be added to testing in the pruner test.

### PR DESCRIPTION
**Pull request by 1maple1**

### CODE REVIEW

File: src/app/beer_garden/app.py [hunk 1]
```diff
@@ -93,39 +93,36 @@ def initialize(self):
...
-            else:
-                self.helper_threads.append(
-                    HelperThread(
-                        db.get_pruner(),
-                        tasks=tasks,
-                        run_every=timedelta(minutes=run_every),
-                        cancel_threshold=ttl_config.in_progress,
-                    )
-                )
+        # Add scheduled jobs for Mongo Pruner
```
Comment (TheBurchLog) [inline src/app/beer_garden/app.py:96]: Should do an overall check if the pruner interval is greater than 0, then do the same for each TTL to determine if it should be scheduled.

File: src/app/beer_garden/db/mongo/pruner.py [hunk 2]
```diff
@@ -6,251 +6,237 @@
...
-    def run(self):
-        """Runs the pruner to delete tasks"""
+            task["collection"].objects(query).no_cache().delete()
 
-        self.logger.debug(
-            "%s is started with run interval %ss", self.display_name, self._run_every
-        )
+    if cancel_threshold > 0:
+        cancel_outstanding(cancel_threshold)
+    logger.debug(display_name + " is stopped")
```
Comment (TheBurchLog) [inline src/app/beer_garden/db/mongo/pruner.py]: OBE logger?

File: src/app/beer_garden/db/mongo/pruner.py [hunk 3]
```diff
@@ -6,251 +6,237 @@
...
                         )
-                        request.delete()
+                    )
+                    task["collection"].objects(query).limit(
+                        task["batch_size"]
+                    ).no_cache().delete()
 
-    def run(self):
-        """Runs the pruner to delete tasks"""
+            task["collection"].objects(query).no_cache().delete()
```
Comment (TheBurchLog) [inline src/app/beer_garden/db/mongo/pruner.py]: For both of the delete commands, we should see if something like this works:
```python
totalDeleted = task["collection"].objects(query).no_cache().delete().count()
logger.info(f"Pruned {totalDeleted} records")
```

Comment (TheBurchLog) [reply src/app/beer_garden/db/mongo/pruner.py]: https://docs.mongoengine.org/apireference.html#mongoengine.Document.delete

File: src/app/beer_garden/db/mongo/pruner.py [hunk 4]
```diff
@@ -6,57 +6,208 @@
...
+                    )
+                    task["collection"].objects(query).limit(
+                        task["batch_size"]
+                    ).no_cache().delete()
+
+            num = task["collection"].objects(query).no_cache().delete()
+            if num:
+                logger.debug(
+                    "Deleted %s %s from %ss"
+                    % (num, ttl_name, task["collection"].__name__)
```
Comment (1maple1) [inline src/app/beer_garden/db/mongo/pruner.py:60]: This only works for files. Delete embedded document in mongoengine is the only delete method that returns a count.

File: src/app/beer_garden/app.py (hunk 1 above)
Comment (1maple1) [reply src/app/beer_garden/app.py:96]: Resolved

File: CHANGELOG.md [hunk 5]
```diff
@@ -7,6 +7,7 @@ TBD
 - Added auto refresh checkbox to requests page
 - Added support for cross-server parent/child requests
 - Reuse existing rabbit admin queue by purging messages instead of creating new queue
+- Refactored Mongo pruner to use APScheduler
```
Comment (TheBurchLog) [inline CHANGELOG.md]: Since we cut the release, need to move this to a new version. Then it is good to go

**Comment by TheBurchLog** [comment]
Need to remove the ttl.multithread option from the config since it shouldn't be used any more
Can you update admin TTL default to 15?

**Comment by 1maple1** [comment]
Removed  ttl.multithread option and updated admin TTL to 15

**Review by TheBurchLog — APPROVED** [review]

**Pull request merged**